### PR TITLE
Updating emitInsCanOnlyWriteSSE2OrAVXReg to cover some of the newer instructions.

### PR DESCRIPTION
**Pull request by tannergooding**
Resolves https://github.com/dotnet/coreclr/issues/19127, https://github.com/dotnet/coreclr/issues/18997, and part of https://github.com/dotnet/coreclr/issues/19008.

### CODE REVIEW

**Comment by tannergooding** [comment]
@dotnet-bot test Windows_NT x64 Checked jitincompletehwintrinsic
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Windows_NT x64 Checked jitnox86hwintrinsic

@dotnet-bot test Windows_NT x86 Checked jitincompletehwintrinsic
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Windows_NT x86 Checked jitnox86hwintrinsic

@dotnet-bot test Ubuntu x64 Checked jitincompletehwintrinsic
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
@dotnet-bot test Ubuntu x64 Checked gcstress0xc
@dotnet-bot test Ubuntu x64 Checked gcstress0xc_minopts_heapverify1

@dotnet-bot test Windows_NT x64 Checked gcstress0xc
@dotnet-bot test Windows_NT x64 Checked gcstress0xc_minopts_heapverify1

**Comment by tannergooding** [comment]
Rebased onto head now that #19130 has been merged.

**Comment by tannergooding** [comment]
@dotnet-bot test Windows_NT x64 Checked jitincompletehwintrinsic
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Windows_NT x64 Checked jitnox86hwintrinsic

@dotnet-bot test Windows_NT x86 Checked jitincompletehwintrinsic
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Windows_NT x86 Checked jitnox86hwintrinsic

@dotnet-bot test Ubuntu x64 Checked jitincompletehwintrinsic
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
@dotnet-bot test Ubuntu x64 Checked gcstress0xc
@dotnet-bot test Ubuntu x64 Checked gcstress0xc_minopts_heapverify1

@dotnet-bot test Windows_NT x64 Checked gcstress0xc
@dotnet-bot test Windows_NT x64 Checked gcstress0xc_minopts_heapverify1

**Comment by tannergooding** [comment]
@dotnet-bot test Windows_NT x64 Checked jitincompletehwintrinsic
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Windows_NT x64 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Windows_NT x64 Checked jitnox86hwintrinsic

@dotnet-bot test Windows_NT x86 Checked jitincompletehwintrinsic
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Windows_NT x86 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Windows_NT x86 Checked jitnox86hwintrinsic

@dotnet-bot test Ubuntu x64 Checked jitincompletehwintrinsic
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnoavx
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
@dotnet-bot test Ubuntu x64 Checked jitx86hwintrinsicnosimd
@dotnet-bot test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
@dotnet-bot test Ubuntu x64 Checked gcstress0xc
@dotnet-bot test Ubuntu x64 Checked gcstress0xc_minopts_heapverify1

@dotnet-bot test Windows_NT x64 Checked gcstress0xc
@dotnet-bot test Windows_NT x64 Checked gcstress0xc_minopts_heapverify1

**Comment by tannergooding** [comment]
CC. @AndyAyersMS, @CarolEidt, @fiigii

File: src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -1259,15 +1259,43 @@ bool emitter::emitInsCanOnlyWriteSSE2OrAVXReg(instrDesc* id)
...
-    if (!IsSSEOrAVXInstruction(ins) || ins == INS_mov_xmm2i || ins == INS_cvttsd2si || ins == INS_cvttss2si ||
-        ins == INS_cvtsd2si || ins == INS_cvtss2si || ins == INS_pmovmskb || ins == INS_pextrw || ins == INS_pextrb ||
-        ins == INS_pextrd || ins == INS_pextrq || ins == INS_extractps)
+    if (!IsSSEOrAVXInstruction(ins))
     {
         return false;
     }
 
-    return true;
+    switch (ins)
```
Comment (tannergooding) [inline src/jit/emitxarch.cpp]: New entries are:
```c++
case INS_andn:
case INS_blsi:
case INS_blsmsk:
case INS_blsr:
case INS_movmskpd:
case INS_movmskps:
case INS_pdep:
case INS_pext:
case INS_pextrw_sse41:
```

File: src/jit/emitxarch.cpp [hunk 2]
```diff
@@ -10099,6 +10127,8 @@ BYTE* emitter::emitOutputAM(BYTE* dst, instrDesc* id, code_t code, CnsVal* addc)
             switch (id->idInsFmt())
             {
                 case IF_RWR_ARD:
+                case IF_RRW_ARD:
+                case IF_RWR_RRD_ARD:
```
Comment (tannergooding) [inline src/jit/emitxarch.cpp]: I wasn't actually getting GCStress failures for the `RWR_RRD_ARD` (as well as the corresponding `SRD`, `MRD`, and `RRD`) entries... However, we are definitely executing SIMD instructions that write integer registers and that use those formats....

Anyone know if there is something I'm missing here?

**Comment by tannergooding** [comment]
As a general comment, when reviewing I saw at least a couple `IF_*` entries that didn't make much sense (such as `IF_RRW_RRW_CNS`)... We should probably audit the instructions and the formats they get mapped to in order to ensure that they make sense.

File: src/jit/emitxarch.cpp (hunk 2 above)
Comment (AndyAyersMS) [reply src/jit/emitxarch.cpp]: You may just be getting (un)lucky, as the written registers won't always contain GC refs.

**Comment by tannergooding** [comment]
@CarolEidt, @AndyAyersMS do either of you have any concerns with merging this?
* The Win x64 HeapVerify/GCStress job timed out, everything passes locally.
* The OSX10.12 failure is in Avx_ro, but is unrelated to this change (still investigating locally).
* The Ubuntu GCStress jobs have been running for 17 hours now (and are still going, they don't appear to be stalled)

**Comment by CarolEidt** [comment]
Added a note to #14523 about this.

**Comment by tannergooding** [comment]
> Might be interesting to see if anything turns up from this outside of methods that use these HW intrinsics.

Good idea, will check this before merging.

**Comment by tannergooding** [comment]
corelib reported 0 diffs. Test diffs are still running.

**Comment by tannergooding** [comment]
Only `Bmi1` reported diffs, and only for the gcinfo portion.

**Comment by AndyAyersMS** [comment]
Thanks for checking. That explains why you didn't see other stress failures ...

**Comment by tannergooding** [comment]
> That explains why you didn't see other stress failures .

Right. They must be getting updated somewhere else, and I missed it somewhere. (I know `RRW_RRW_RRD` was handle explicitly after the `emitOutputRR` call, for example).

**Pull request merged**